docs/building-blocks/extending-templates/collector.md

```diff
@@ -0,0 +1,308 @@
+---
+title: Implement a property transformer and collector in an Azure Resource Manager template
+description: Describes how to implement a property transformer and collector in an Azure Resource Manager template
+author: petertay
+ms.date: 06/09/2017
+
+---
+
+# Implement a property transformer and collector in an Azure Resource Manager template
+
+In [use an object as a parameter in an Azure Resource Manager template][objects-as-parameters], you learned how to store resource property values in an object and apply them to a resource during deployment. While this is a very useful way to manage your parameters, it still requires you to map the object's properties to resource properties each time you use it in your template.
+
+To work around this, you can implement a property transform and collector that iterates your object array and transforms it into the JSON schema expected by the resource.
+
+> [!IMPORTANT]
+> This approach requires that you have a deep understanding of Resource Manager templates and functions.
+
+Let's take a look at how we can implement a property collector and transformer with an example that deploys a [network security group (NSG)][nsg]. The diagram below shows the relationship between our templates and our resources within those templates:
+
+![property collector and transformer architecture](../_images/collector-transformer.png)
+
+Our **calling template** includes two resources:
+* a template link that invokes our **collector template**.
+* an NSG.
+
+Our **collector template** includes two resources:
+* an **anchor** resource.
+* a template link that invokes the transform template in a copy loop.
+
+Our **transform template** includes a single resource: an empty template with a variable that transforms our `source` JSON to the JSON schema expected by our NSG resource in the **main template**.
+
+## Parameter object
+
```

Comment:
BTW, this was the winning paragraph for me. I only understood what we were trying to do when I got here. I suggest maybe moving the first diagram and list of elements to after this example.